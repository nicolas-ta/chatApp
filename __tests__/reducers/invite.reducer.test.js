import {inviteReducer} from '@reducers/invite';

const mockUser1 = {userId: 'test1'};
const mockUser2 = {userId: 'test2'};

describe('Invite reducer unit tests', () => {
  test('checking initial state', () => {
    const action = {type: 'nonexistent'};
    expect(inviteReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking refresh state', () => {
    const action = {type: 'refresh'};
    expect(inviteReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking reset-selection state', () => {
    const action = {type: 'reset-selection'};
    expect(inviteReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking fetch-users empty state', () => {
    const action = {
      type: 'fetch-users',
    };
    const state = {
      channel: undefined,
      users: [mockUser1],
      userMap: {userId: 'test2'},
    };
    expect(inviteReducer(state, action)).toMatchSnapshot();
  });

  test('checking fetch-users state', () => {
    const action = {
      type: 'fetch-users',
      payload: {users: [{userId: 'test1'}, {userId: 'test2'}]},
    };
    const state = {
      channel: {members: [mockUser1]},
      users: [mockUser1],
      userMap: {userId: 'test2'},
    };
    expect(inviteReducer(state, action)).toMatchSnapshot();
  });

  test('checking select-user empty state', () => {
    const action = {type: 'select-user'};
    const state = {
      error: '',
      selectedUsers: [mockUser1, mockUser2],
    };
    expect(inviteReducer(state, action)).toMatchSnapshot();
  });

  test('checking select-user state', () => {
    const action = {type: 'select-user', payload: {user: mockUser1}};
    const state = {
      error: '',
      selectedUsers: [mockUser1, mockUser2],
    };
    expect(inviteReducer(state, action)).toMatchSnapshot();
  });

  test('checking select-user not included state', () => {
    const action = {type: 'select-user', payload: {user: {userId: 'test3'}}};
    const state = {
      error: '',
      selectedUsers: [mockUser1, mockUser2],
    };
    expect(inviteReducer(state, action)).toMatchSnapshot();
  });

  test('checking unselect-user empty state', () => {
    const action = {type: 'unselect-user'};
    const state = {
      error: '',
      selectedUsers: [mockUser1, {userId: 'test1'}],
    };
    expect(inviteReducer(state, action)).toMatchSnapshot();
  });

  test('checking unselect-user included state', () => {
    const action = {type: 'unselect-user', payload: {user: mockUser1}};
    const state = {
      error: '',
      selectedUsers: [mockUser1, {userId: 'test1'}],
    };
    expect(inviteReducer(state, action)).toMatchSnapshot();
  });

  test('checking unselect-user not included state', () => {
    const action = {type: 'unselect-user', payload: {user: mockUser1}};
    const state = {
      error: '',
      selectedUsers: [mockUser2, {userId: 'test3'}],
    };
    expect(inviteReducer(state, action)).toMatchSnapshot();
  });

  test('checking start-loading state', () => {
    const action = {type: 'start-loading'};
    expect(inviteReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking end-loading state', () => {
    const action = {type: 'end-loading'};
    expect(inviteReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking error state', () => {
    const action = {type: 'error'};
    expect(inviteReducer(undefined, action)).toMatchSnapshot();
  });
});
