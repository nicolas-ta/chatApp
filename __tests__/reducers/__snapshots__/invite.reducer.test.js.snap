// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invite reducer unit tests checking end-loading state 1`] = `
Object {
  "error": null,
  "loading": false,
}
`;

exports[`Invite reducer unit tests checking error state 1`] = `
Object {
  "error": undefined,
}
`;

exports[`Invite reducer unit tests checking fetch-users empty state 1`] = `
Object {
  "channel": undefined,
  "userMap": Object {
    "test1": true,
  },
  "users": Array [
    Object {
      "userId": "test1",
    },
  ],
}
`;

exports[`Invite reducer unit tests checking fetch-users state 1`] = `
Object {
  "channel": Object {
    "members": Array [
      Object {
        "userId": "test1",
      },
    ],
  },
  "userMap": Object {
    "test1": true,
    "test2": true,
  },
  "users": Array [
    Object {
      "userId": "test1",
    },
    Object {
      "userId": "test2",
    },
  ],
}
`;

exports[`Invite reducer unit tests checking initial state 1`] = `undefined`;

exports[`Invite reducer unit tests checking refresh state 1`] = `
Object {
  "error": "",
  "loading": false,
  "selectedUsers": Array [],
  "userMap": Object {},
  "users": Array [],
}
`;

exports[`Invite reducer unit tests checking reset-selection state 1`] = `
Object {
  "loading": false,
  "selectedUsers": Array [],
}
`;

exports[`Invite reducer unit tests checking select-user empty state 1`] = `
Object {
  "error": "",
  "selectedUsers": Array [
    Object {
      "userId": "test1",
    },
    Object {
      "userId": "test2",
    },
    undefined,
  ],
}
`;

exports[`Invite reducer unit tests checking select-user not included state 1`] = `
Object {
  "error": "",
  "selectedUsers": Array [
    Object {
      "userId": "test1",
    },
    Object {
      "userId": "test2",
    },
    Object {
      "userId": "test3",
    },
  ],
}
`;

exports[`Invite reducer unit tests checking select-user state 1`] = `
Object {
  "error": "",
  "selectedUsers": Array [
    Object {
      "userId": "test1",
    },
    Object {
      "userId": "test2",
    },
  ],
}
`;

exports[`Invite reducer unit tests checking start-loading state 1`] = `
Object {
  "error": null,
  "loading": true,
}
`;

exports[`Invite reducer unit tests checking unselect-user empty state 1`] = `
Object {
  "error": "",
  "selectedUsers": Array [
    Object {
      "userId": "test1",
    },
    Object {
      "userId": "test1",
    },
  ],
}
`;

exports[`Invite reducer unit tests checking unselect-user included state 1`] = `
Object {
  "error": "",
  "selectedUsers": Array [],
}
`;

exports[`Invite reducer unit tests checking unselect-user not included state 1`] = `
Object {
  "error": "",
  "selectedUsers": Array [
    Object {
      "userId": "test2",
    },
    Object {
      "userId": "test3",
    },
  ],
}
`;
